docs/assignment2.md

```diff
@@ -1 +1,233 @@
+
+#### **GamePane**
+
+Abstract methods
+
+- *void connectComponents()*: Connects all components into the root of a *javafx.scene.Scene*
+- *void styleComponents()*: Styles all components as required
+- *void setCallbacks()*: Sets callbacks for all interactive components
+
+### *Controllers* package
+
+Only consists of class **SceneManager** used for managing the *javafx* Scenes.
+
+Operations
+
+- *void setStage(final Stage stage)*: Sets the primary stage
+- *void showScene(final Scene scene)*: Replaces the currently active Scene with another one
+- *void showPane(final Class<? extends GamePane> pane)*: Replaces the current  **GamePane** with another
+- *SceneManager getInstance()*: returns the instance of the class
+
+
+## Object diagrams								
+Author(s): `Torstein Thomassen`
+
+This object diagram shows an example of objects with different states in our program.
+All the different attributes and variables has to be fulled in with example data.
+
+In this example each team has only three players. This is only for demonstration and will be changed for the final version.
+
+This structure resembles a tree structure with branches and leafs. The tree structure is commonly used in many software design. It's an efficient and a good way to structure the relation between objects. 
+
+This creates a hierarchy where a parent can only directly affect the children. E.g the **Match** can't affect the **Player** 'directly', only the **Team** objects.       
+
+`Object diagram over the Match Class`
+![](imgs/ass2%20obj%20diagram.png)
+
+The diagram above shows an example over the *Match* class. It has a relation with two *Team* object, and the *Team* object again has a relation with various *Player* object
+As you can see *Team 1* would win the match because they have a higher score. You can also see it's the second round, because both teams have already one win.  
+
+
+
+## State machine diagrams									
+Author(s): `Torstein Thomassen` & `Teona Moga`
+
+
+
+In the way our system is designed, only the **Game** and **League** classes are not stateless. Their state diagrams are shown below.
+
+`State diagram over League Class`
+![](imgs/StatechartDiagram1.jpg)
+
+The state diagram above represents the different states a league can be in. When it is initialized, no rounds are played so the *winners* map is empty. **League** has one recursive method, called when the results of the created schedule, i.e. **initialTeams**, are requested. What breaks the recursion is having no more matches to play in a subsequent round, as there is only one team left (the winner of the league).
+
+`State diagram over Game Class`
+![](imgs/ass2%20state%20Diagram.png)
+
+Above is shown a state diagram for the **Game** class. You have different methods that effect the three main states the **Game** class consist of.
```

Comment:
**[SEVERE]** The Decision node here in unsuitable, you probably needed  a parrallelization node along with 3 composite states for each pair of corresponding states.

As it stands, this diagram is the bare minimum!